test/functional/mempool_persist.py

```diff
@@ -80,6 +83,11 @@ def run_test(self):
         assert_greater_than_or_equal(tx_creation_time, tx_creation_time_lower)
         assert_greater_than_or_equal(tx_creation_time_higher, tx_creation_time)
 
+        # disconnect nodes & make a txn that remains in the unbroadcast set.
+        disconnect_nodes(self.nodes[0], 2)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), Decimal("12"))
+        connect_nodes(self.nodes[0], 2)
```

Comment:
If I add `time.sleep(10)` before or after `connect_nodes`, then the test fails locally for me (the mempool for node 2 has 6, instead of the expected 5 transactions).

Does this test depend on particular timings (e.g. node 1 doesn't have time to relay the transaction to node 2 before the nodes are all shut down)?  If so, in a future change would it make sense to either (i) document the necessary timing assumptions, or (ii) update the test to depend less on particular timings?

Reply:
good catch! this is a bug. line 87 isn't actually disconnecting the two nodes because its passing `node_num` of `2` rather than `1`

I've added commit 41fd9c1b7b372d7769e005db1d92000c0373651e on the follow-up PR #18807 to fix this issue.

thank you!

Reply:
Confirmed that with 41fd9c1b7b372d7769e005db1d92000c0373651e the test passes even with random `time.sleep(10)` thrown.  Thanks for updating!